packages/utils/plugins-helpers/src/federation.ts

```diff
@@ -30,14 +30,17 @@ export const federationSpec = parse(/* GraphQL */ `
 `);
 
 /**
- * Adds `__resolveReference` in each ObjectType involved in Federation.
+ * Adds `__resolveReference` in each ObjectType involved in Federation, only when there's a mapped type in config.mappers.
  * @param schema
  */
-export function addFederationReferencesToSchema(schema: GraphQLSchema): GraphQLSchema {
+export function addFederationReferencesToSchema(
+  schema: GraphQLSchema,
+  mappersTypeName: string[] | undefined
+): GraphQLSchema {
   const typeMap = schema.getTypeMap();
   for (const typeName in typeMap) {
     const type = schema.getType(typeName);
-    if (isObjectType(type) && isFederationObjectType(type)) {
+    if (isObjectType(type) && isFederationObjectType(type) && mappersTypeName && mappersTypeName.includes(typeName)) {
```

Comment:
again, `__resolveReference` is used to resolve an a reference to an entity which means mappers can't change that behavior. Mappers are only for "us" to define an exact type instead of the default one from the schema.